teps/0062-catalog-tags-and-hub-categories-management.md

```diff
@@ -0,0 +1,195 @@
+    tekton.dev/category: "foo”     👈 Category of a resource
+    tekton.dev/tags: python, pylint   👈 New tags
+    tekton.dev/displayName: pylint
+```
+
+- CI/Catlin should make sure that category is added for the task and there is only one category. Maintaining the category is the responsibility of the author.
+- If category is not added,Catlin should throw an warning to add a category
+- The category should be added from the predefined category list
+
+  - Sample Predefined Category List:
+
+  ```yaml
+  categories:
+    - Automation
+    - Build Tools
+    - CLI
+    - Cloud
+    - Deploy
+    - Editor
+    - Git
+    - Image Build
+    - Language
+    - Messaging
+    - Monitoring
+    - Notification
+    - Others
+    - Security
+    - Storage
+    - Test Framework
+  ```
+
+- If users wants to add new categories to the above list, the user should create a pull request to [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml)
+
+- Tags can be used in hub as keywords and resources can be searched using tags and can be filtered only by categories
+
+  For e.g: Search in github pr’s (here one of the keyword can be like `is:pr is:open`)
+
+  Example of Operator Hub
+
+  Yes, Operator Hub provides both categories and tags for CSV. The word they use is keywords for tags.
+
+1. Predefined list of categories for Operator Hub - <https://github.com/operator-framework/api/blob/master/pkg/validation/internal/operatorhub.go#L37>
+
+2. Choosing a single category in CSV - <https://github.com/operator-framework/community-operators/blob/834f2e89f3f73612cc59496c113ce21d739d991f/community-operators/etcd/0.9.4/etcdoperator.v0.9.4.clusterserviceversion.yaml#L19>
+
+3. Specifying keywords(tags) in CSV - <https://github.com/operator-framework/community-operators/blob/834f2e89f3f73612cc59496c113ce21d739d991f/community-operators/etcd/0.9.4/etcdoperator.v0.9.4.clusterserviceversion.yaml#L282>
+
+- Steps to be taken while following this solution
+
+  - Add categories to existing tasks in catalog
+  - Add Catlin in catalog CI and further the category check
+  - Update the catalog’s pr template and contribution docs
+  - Add a doc to guidelines on adding a new category
+  - Update [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml) in Hub repo
+  - Update search in Hub, so that resources can be searched by tags as well
+
+### Alternative solution
+
+- User can create a pull request to Hub by making changes in [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml) to add tags to the corresponding categories or add new categories - details can be provided in a doc of hub repo
+- Add information in Catalog’s pr template to showcase how one can add categories and tags to hub with reference to hub doc
+- Add a doc in catalog repo either in contribution doc or somewhere
+- Catlin to show some recommendations whenever there is a pr on Catalog. Like it may do some comment on the PR if found a tag not mapping to any category.
+
```

Comment:
one other alternative is to not have categories all together

Reply:
So it would affect the Hub UI as well as tags will be endless, so there would be a list of many tags on the left side where categories are and there can be tags such as `git, github`, etc which would have same meaning but different names and as a UX point of view it would affect the user experience as well. Since categories is a short generalized list which would map all tasks hence the filtering would be easy and tags can be search as keywords in Hub